### ISSUE TITLE
ASLMap.java & ASLBoardPicker cleanup

### ISSUE THREAD

**Issue by derimmer** [vasl-developers/vasl#1521]
@uckelman did a great job of cleaning up the ASLChatter code. There are lots of other classes that could benefit from such work. I am going to try to do 1 or 2 each version, starting with ASLMap and ASLBoardPicker in 668.

**Comment by derimmer** [comment]
I am starting on the cleanup of ASLBoardPicker. I am emulating the work that @uckelman did on ASLChatter, essentially looking at every commit he pushed and trying to make the same changes (if required) in ASLBoardPicker. Commit c6b6a41 on Aug 27, 2023, named "Removed unnecessary this" is self-explanatory. 

In ASLBoardPicker, the following lines of code appear 3 times:

`updateBoard(name);
 this.tileBoard(name);`

These methods appear to me to have the same signature (if that is the right term):

`private void updateBoard(String baseName) {`
`private void tileBoard(String commonName) {`

It seems to me that `this.tileBoard(name);` qualifies as an 'unnessecary this" and should be removed in each instance. Am I understanding this correctly?

**Comment by uckelman** [comment]
The only time you ever need `this` in Java is when the thing after the `this` would be ambiguous without it, such as when you have a function parameter that shadows a member.

**Comment by uckelman** [comment]
Any variable that the compiler will let you make `final` you should make `final`. It's unfortunate that in Java, `final` was not made the default.

The only reason for not making a variable `final` is if _you actually reassign it_.

**Comment by derimmer** [comment]
I have completed the code cleanup of ASLBoardPicker. I suspect I missed a number of things but I wanted to be sure I didn't break anything. So far the class continues to work.

**Comment by derimmer** [comment]
I have just pushed the changes to ASLMap and several related class to complete the ASLMap cleanup. I did a much better job with this class than with ASLBoardPicker. I was able to see more of the issues and fix them. 

I am going to close this issue and create a new one for the next two classes that I am going to clean up: VASL.LOS.MAP.Map and VASLThread, both of which I have amended extensively over the years and now need a cleanup.

I am also going to create an issue to Refactor ASLMaps. In particular all of the code that I added (literally hundreds of lines of code) to enable los to work with traditional overlays need to be refactored to remove duplication. Given the complexity of the task (having to account for cropped and flipped maps), I chose to code each variation separately and these can now be better integrated. Plus, since I did this overlay type-by-type, I had some test for overlay type and these again can be removed or consolidated.
